resources/mediaplayer/src/main/java/org/mobicents/media/server/impl/resource/mediaplayer/audio/CachedRemoteStreamProvider.java

```diff
@@ -0,0 +1,79 @@
+package org.mobicents.media.server.impl.resource.mediaplayer.audio;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+import org.apache.commons.io.IOUtils;
+import org.ehcache.Cache;
+import org.ehcache.CacheManager;
+import org.ehcache.config.builders.CacheConfigurationBuilder;
+import org.ehcache.config.builders.CacheManagerBuilder;
+import org.ehcache.config.builders.ResourcePoolsBuilder;
+import org.ehcache.config.units.MemoryUnit;
+
+/**
+ * Created by achikin on 5/9/16.
+ */
+public class CachedRemoteStreamProvider implements RemoteStreamProvider {
+
+    private CacheManager cacheManager;
+
+    public CachedRemoteStreamProvider(int size) {
+        cacheManager = CacheManagerBuilder.newCacheManagerBuilder()
+                .withCache("preConfigured",
+                        CacheConfigurationBuilder.newCacheConfigurationBuilder(URL.class, AudioStreamCache.class,
+                                ResourcePoolsBuilder.newResourcePoolsBuilder().heap(size, MemoryUnit.MB))
```

Comment:
It would be nice to have an option for disk caches. Just using the builder with different option http://www.ehcache.org/apidocs/3.1.0/org/ehcache/config/builders/ResourcePoolsBuilder.html#disk-long-org.ehcache.config.units.MemoryUnit-boolean-